### PR TITLE
choice(ndlarray)

### PR DESCRIPTION
**Pull request by gdementen**
added support for choice with an ND LArray

### CODE REVIEW

**Comment by gdementen** [comment]
@alixdamman I wonder if it is worth a mention in the release notes. I developed this because I know I will need this in LIAM2 at some point (not directly though) but in the general case, I don't think any of our current users cares about this.

File: larray/random.py [hunk 1]
```diff
@@ -379,33 +379,34 @@ def choice(choices=None, axes=None, replace=True, p=None, meta=None):
         Otherwise, if the resulting axes have a shape of, e.g., ``(m, n, k)``, then ``m * n * k`` samples are drawn.
     replace : boolean, optional
         Whether the sample is with or without replacement.
-    p : 1-D array-like, optional
+    p : array-like, optional
         The probabilities associated with each entry in choices.
-        If p is a 1-D LArray, choices are taken from its axis.
+        If p is an LArray, choices are taken from its axes. If p is N dimensional, each cell represents the probability
+        that the combination of labels will occur.
         If not given the sample assumes a uniform distribution over all entries in choices.
```
Comment (alixdamman) [inline larray/random.py]: I wonder if it wouldn't be better to say "If p is a 1-D LArray, choices are taken from its axis. If p is N dimensional, each cell represents the probability that the combination of labels will occur."
I don't know.

File: larray/random.py [hunk 2]
```diff
@@ -446,16 +447,77 @@ def choice(choices=None, axes=None, replace=True, p=None, meta=None):
...
+        else:
+            choices = p.axes[0].labels
+            p = p.data
     if choices is None:
         raise ValueError("choices argument must be provided unless p is an LArray")
     return LArray(np.random.choice(choices, axes.shape, replace, p), axes, meta=meta)
+
+
+# XXX: make this into a method or property on AxisCollection? axes.flat_labels[flat_indices]?
+def axes_flat_lookup(axes, flat_indices):
```
Comment (alixdamman) [inline larray/random.py]: at least move this function to core/axis.py or make it a private method of AxisCollection.

**Comment by alixdamman** [comment]
I also think there is no need to mention it in the release notes.

File: larray/random.py (hunk 1 above)
Comment (gdementen) [reply larray/random.py]: I hesitated with that too. It seemed redundant, but is probably easier to understand, so ok, I'll change that.

File: larray/random.py (hunk 2 above)
Comment (gdementen) [reply larray/random.py]: ok

File: larray/core/axis.py [hunk 3]
```diff
@@ -2641,6 +2641,48 @@ def align(self, other, join='outer', axes=None):
...
         return self.replace(axes_changes), other.replace(axes_changes)
 
+    # XXX: make this into a public method/property? axes_col.flat_labels[flat_indices]?
+    def _flat_lookup(self, flat_indices):
+        r"""Return labels corresponding to indices into the flattened axes
+
+        Parameters
+        ----------
+        axes : AxisCollection-like
+            axes to look up into.
```
Comment (alixdamman) [inline larray/core/axis.py]: There is no `axes` argument anymore

Comment (gdementen) [reply larray/core/axis.py]: doh! ☹️

**Comment by gdementen** [comment]
@alixdamman want to review once more or do I just merge this as is? (I only removed the docstring lines)

**Comment by alixdamman** [comment]
Merge

**Pull request merged**